### PR TITLE
fix(string/truncate): don't replace single-character strings with the ellipsis

### LINKED ISSUE #558: Single-character strings are pointelessly "truncated"
### Pre-Checks

- [x] Follow our [Code of Conduct](https://github.com/visulima/visulima/blob/main/.github/CODE_OF_CONDUCT.md).
- [x] Read the [Contributing Guidelines](https://github.com/visulima/visulima/blob/main/.github/CONTRIBUTING.md).
- [x] Check that there isn't [already an issue](https://github.com/visulima/visulima/issues) that reports the same bug to avoid creating a duplicate.
- [x] Make sure this is a Visulima issue and not related to a combination with another package.
- [x] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/visulima/visulima/discussions) or join our [Discord Chat Server](https://chat.visulima.com).
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.
- [x] I am willing to provide a PR.

### Describe the bug

Strings that are just a single character long are replaced with the truncation character, instead of left as-is

### Minimal reproduction code

```ts
console.log(truncate("+", 1));
```

#### Output

```
…
```

### Additional Context

_No response_

### Environment Info

```shell
`@visulima/string` version 2.0.1
```

### Which module system do you use?

- [ ] CJS
- [x] ESM

### Used Package Manager

yarn

### PR DESCRIPTION
**Pull request by ItsHarper**
Fixes #558 

The `tabular` package needs to be updated alongside `string` so that it gets this fix too (probably applies to others as well, but that's the one I care about).

## Summary by CodeRabbit

* **Tests**
  * Enhanced test coverage for string truncation edge cases.

* **Refactor**
  * Optimized internal logic flow in truncation functionality.

### CODE REVIEW

File: packages/string/__tests__/unit/truncate.test.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ describe(truncate, () => {
             expect(truncate("unicorn", 20)).toBe("unicorn");
             expect(truncate("unicorn", 7)).toBe("unicorn");
             expect(truncate("unicorn", 6)).toBe("unico…");
+            expect(truncate("y", 1)).toBe("y");
```
Comment (prisis) [inline packages/string/__tests__/unit/truncate.test.ts]:
```suggestion

            expect(truncate("y", 1)).toBe("y");
            expect(truncate("ye", 1)).toBe("…");
            expect(truncate("yes", 1)).toBe("…");
```

**Comment by ItsHarper** [comment]
Seems like your merge broke CI

**Comment by prisis** [comment]
Because of the monorepo, i have a small problem with "cerebro" its still a cjs/esm package, but i did migrate pail, fixed it for now, so the pipelines will run

**Comment by prisis** [comment]
@ItsHarper fix was released with tabular and string 2.0.2

**Comment by ItsHarper** [comment]
@prisis Thanks! Unfortunately, I'm still seeing my 1-character table header get replaced with an ellipsis.

**Comment by prisis** [comment]
Can you provide me some code so I have a reproduction?

It bundles only the part that is used otherwise the string package is a bit big because of all the i18n support in some functions

**Comment by prisis** [comment]
I think i found the issue, it in the grid.ts:1004 on tabular. Need to check what is the best way to fix the issue, looks like i need to recheck the baseContentWidth...
